extensions/kinesis/src/main/java/com/hazelcast/jet/kinesis/KinesisSources.java

```diff
@@ -152,10 +158,13 @@ public static Builder kinesis(@Nonnull String stream) {
         @Nonnull
         private RetryStrategy retryStrategy = DEFAULT_RETRY_STRATEGY;
         @Nonnull
-        private InitialShardIterators initialShardIterators = new InitialShardIterators();
+        private final InitialShardIterators initialShardIterators = new InitialShardIterators();
+        @Nonnull
+        private BiFunctionEx<? super Record, ? super Shard, ? extends T> projectionFn;
 
         private Builder(@Nonnull String stream) {
             this.stream = stream;
+            this.projectionFn = (record, shard) -> (T) toArray(record, shard);
```

Comment:
```suggestion
            this.projectionFn = (record, shard) -> (T) entry(record.getPartitionKey(), toArray(record, shard));
```

In order to keep the old default behavior same, we need to change this `projectionFn` to the suggested one (don't forget to add import static `com.hazelcast.jet.Util.entry`).

Reply:
Good catch!